pushapkscript/src/pushapkscript/script.py

```diff
@@ -29,18 +29,26 @@ async def async_main(context):
 
     all_apks_paths = [artifact for artifacts_list in artifacts_per_task_id.values() for artifact in artifacts_list if artifact.endswith(".apk")]
 
+    all_aabs_paths = [artifact for artifacts_list in artifacts_per_task_id.values() for artifact in artifacts_list if artifact.endswith(".aab")]
+
     if not publish_config.get("skip_check_signature", True):
         log.info("Verifying APKs' signatures...")
         for apk_path in all_apks_paths:
             jarsigner.verify(context, publish_config, apk_path)
             manifest.verify(product_config, apk_path)
+        log.info("Verifying AABs' signatures...")
+        for aab_path in all_aabs_paths:
+            jarsigner.verify(context, publish_config, aab_path)
+            manifest.verify(product_config, aab_path)
     else:
-        log.info('This product is configured with "skip_check_signature", so the signing of the ' "APK will not be verified.")
+        log.info('This product is configured with "skip_check_signature", so the signing of the ' "files will not be verified.")
 
     log.info("Delegating publication to mozapkpublisher...")
     with contextlib.ExitStack() as stack:
         files = [stack.enter_context(open(apk_file_name)) for apk_file_name in all_apks_paths]
         publish.publish(product_config, publish_config, files, contact_server)
+        files = [stack.enter_context(open(aab_file_name)) for aab_file_name in all_aabs_paths]
+        publish.publish_aab(product_config, publish_config, files, contact_server)
```

Comment:
Question: What's the reasoning for having a single pushapk task that handles both APKs and the AAB? If I understand correctly, switching to AABs is permanent so we won't have both at the same time. I wonder if this could cause confusion for people who have little context on APKs vs AABs. What do you think?

Reply:
Doing it this way delegates the issue to the task configuration: We can have one task with both APK and AAB upstream artifacts, or separate tasks with only APK (or AAB) upstream artifacts; pushapkscript will just push whatever artifacts are configured. I suppose it is odd to have "pushapkscript" pushing AABs but I don't think it is worth renaming everything now.

Reply:
Thanks for the context!

> We can have one task with both APK and AAB upstream artifacts, [...] pushapkscript will just push whatever artifacts are configured.

If I'm not mistaken, this case won't work, right? If a product turned AABs on, then Google Play will refuse any APKs submitted, won't it? (Just making sure I didn't miss anything obvious 🙂)

Reply:
Right! We should configure tasks to push APKs or AABs, but not both.

...so let's add a check that no one tries to do both...